rct229/rulesets/ashrae9012019/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/aggregate_min_OA_schedule_across_zones.py

```diff
@@ -0,0 +1,31 @@
+from operator import add
+
+
+def aggregate_min_OA_schedule_across_zones(zone_OA_CFM_list_of_schedules):
+    """Each HVAC system can serve multiple zones which can have multiple terminal units serving each zone. In order to obtain an hourly HVAC level OA CFM schedule the OA CFM provided by each zone needs to be aggregated for each hour of the year. This function receives an RMR (B, U, or P) and a list of zone OA min cfm schedules (generated by the get_min_OA_CFM_sch_zone function) and loops through each zone to create an aggregated 8760 for OA CFM for the HVAC system.
+
+    Parameters
+    ----------
+    zone_OA_CFM_list_of_schedules: list of list
+        A list which contains each zone's 8760 OA CFM schedule as a list value.
+
+    Returns
+    -------
+    list
+        This function receives an RMR (B, U, or P) and a list of zone OA min cfm schedules (generated by the get_min_OA_CFM_sch_zone function) and loops through each zone to create an aggregated 8760 for OA CFM for the HVAC system.
+    """
+
+    assert all(
+        [
+            len(zone_OA_CFM_list_of_schedule) in (8760, 8784)
+            for zone_OA_CFM_list_of_schedule in zone_OA_CFM_list_of_schedules
+        ]
+    ), f"Not all schedules are either length of 8760 or 8784."
```

Comment:
Let's use `assert_` instead of `assert` because `assert_` raise to an exception that can be handled by engine.

In this assertion, we shall check two things: 1. len of a list is either 8760 or 8784, 2. all lists shall have same length.

You can use iter to do the check
```python
schedule_iter = iter(zone_OA_CFM_list_of_schedule)
schedule_len = len(next(schedule_iter))
assert_(all(len(schedule) == schedule_len and len(schedule) in (8760, 8784) for schedule in schedule_iter)), "Not all schedules...")

```

Reply:
Good point! addressed.